apps/aecore/src/aec_peer_connection.erl

```diff
@@ -711,23 +751,57 @@ handle_get_block_rsp(S, {get_block, From, _TRef}, Msg) ->
     end,
     remove_request_fld(S, get_block).
 
-%% -- Get Mempool --------------------------------------------------------------
+%% -- TX Pool ----------------------------------------------------------------
+
+handle_tx_pool_sync_init(S, _MsgObj) ->
+    case aec_tx_pool_sync:accept(peer_id(S), self()) of
+        ok ->
+            send_response(S, txps_init, {ok, #{}});
+        Err = {error, _} ->
+            send_response(S, txps_init, Err)
+    end,
+    S.
+
+handle_tx_pool_sync_unfold(S, MsgObj) ->
+    Unfolds = maps:get(unfolds, MsgObj, []),
+    case aec_tx_pool_sync:sync_unfold(peer_id(S), Unfolds) of
+        {ok, NewUnfolds} ->
+            send_response(S, txps_unfold, {ok, #{ unfolds => NewUnfolds }});
+        Err = {error, _} ->
+            send_response(S, txps_unfold, Err)
+    end,
+    S.
 
-handle_get_mempool(S, _MsgObj) ->
-    {ok, Txs0} = aec_tx_pool:peek(infinity),
-    Txs = [ aetx_sign:serialize_to_binary(T) || T <- Txs0 ],
-    send_response(S, mempool, {ok, #{ txs => Txs }}),
+handle_tx_pool_sync_get(S, MsgObj) ->
+    TxHashes = maps:get(tx_hashes, MsgObj, []),
+    case aec_tx_pool_sync:sync_get(peer_id(S), TxHashes) of
+        {ok, Txs} ->
+            SerTxs = [ aetx_sign:serialize_to_binary(Tx) || Tx <- Txs ],
+            send_response(S, txs, {ok, #{ txs => SerTxs }});
+        Err = {error, _} ->
+            send_response(S, txs, Err)
+    end,
     S.
 
-handle_get_mempool_rsp(S, {get_mempool, From, _TRef}, MsgObj) ->
-    case maps:get(txs, MsgObj, undefined) of
-        Txs0 when is_list(Txs0) ->
-            Txs = [ aetx_sign:deserialize_from_binary(T) || T <- Txs0 ],
+handle_tx_pool_sync_finish(S, MsgObj) ->
+    aec_tx_pool_sync:sync_finish(peer_id(S), {done, maps:get(done, MsgObj, false)}),
+    send_response(S, txps_finish, {ok, #{ done => maps:get(done, MsgObj, false) }}),
+    S.
+
+handle_tx_pool_sync_rsp(S, Action, {tx_pool, From, _TRef}, MsgObj) ->
+    case Action of
+        init ->
+            gen_server:reply(From, ok);
+        unfold ->
+            gen_server:reply(From, {ok, maps:get(unfolds, MsgObj, [])});
```

Comment:
Why default `[]`? Is this not guaranteed by https://github.com/aeternity/epoch/pull/1245/files#diff-c1692ca9a93deb9e4ca76237ebbc01fdR217 ? I find that unnecessary default hinders readability (though I understand may be debatable).